### PR TITLE
Fix header/footer temp file is removed before wkhtmltopdf called

### PR DESCRIPTION
**Pull request by thanhcuong1990**
Fix issue #1031. 

Header/footer generated by `WickedPdf::Tempfile` is deleted before wkhtmltopdf called. 
This PR try to use normal `File` then delete it when the pdf file generated. 

Temporary fix in your Gemfile.
`gem 'wicked_pdf', git: 'https://github.com/thanhcuong1990/wicked_pdf.git', branch: 'master'`

### CODE REVIEW

**Review by andrebnf — APPROVED** [review]
Fixed bug in production environment as listed here: https://github.com/mileszs/wicked_pdf/issues/1031#issuecomment-1287080936

Edit @ 2023-01-08 - forked version working fine since the original comment date

**Comment by iggant** [comment]
hi, we really need this PR in the gem. Can we please merge it and release new version?

File: lib/wicked_pdf.rb [hunk 1]
```diff
@@ -61,9 +62,8 @@ def pdf_from_url(url, options = {})
     if track_progress?(options)
       invoke_with_progress(command, options)
     else
-      err = Open3.popen3(*command) do |_stdin, _stdout, stderr|
-        stderr.read
-      end
+      _out, err, status = Open3.capture3(*command)
+      err = [status.to_s, err].join("\n") if !err.empty? || !status.success?
```
Comment (unixmonkey) [inline lib/wicked_pdf.rb:66]: Why the change from `popen3` to `capture3` here, and the joining?

File: lib/wicked_pdf.rb [hunk 2]
```diff
@@ -79,6 +79,7 @@ def pdf_from_url(url, options = {})
   rescue StandardError => e
     raise "Failed to execute:\n#{command}\nError: #{e}"
   ensure
+    clean_temp_files
```
Comment (unixmonkey) [inline lib/wicked_pdf.rb:82]: 👍

File: lib/wicked_pdf/option_parser.rb [hunk 3]
```diff
@@ -61,7 +61,7 @@ def parse_header_footer(options)
           r += make_options(opt_hf, [:line], hf.to_s, :boolean)
           if options[hf] && options[hf][:content]
             @hf_tempfiles = [] unless defined?(@hf_tempfiles)
-            @hf_tempfiles.push(tf = WickedPdf::Tempfile.new("wicked_#{hf}_pdf.html"))
+            @hf_tempfiles.push(tf = File.new(Dir::Tmpname.create(["wicked_#{hf}_pdf", '.html']) {}, 'w'))
```
Comment (unixmonkey) [inline lib/wicked_pdf/option_parser.rb:64]: Is there a reason `WickedPdf::Tempfile` doesn't work here? I remember adding that because an earlier version of Ruby didn't allow tempfiles to be created with an extension, which I know is no longer the case, but don't remember when that changed.

**Review by unixmonkey — COMMENTED** [review]
This is great! Sorry it took a bit for me to take a closer look.

I did post a few questions if you don't mind answering.

File: lib/wicked_pdf/option_parser.rb (hunk 3 above)
Comment (thanhcuong1990) [reply lib/wicked_pdf/option_parser.rb:64]: @unixmonkey `WickedPdf::Tempfile` still can created the temp file. But the files is deleted before the pdf file generated.

**Comment by iggant** [comment]
@unixmonkey please, can you review and merge this PR?

**Comment by unixmonkey** [comment]
This is now in version 2.7.0 of wicked_pdf. Thank you for your help!
